implementation of stack

Rasingan opened the pull request:
Signed-off-by: Rasingan <siddheshrainbow@gmail.com>

Rasingan committed:
implementation of stack
Signed-off-by: Rasingan <siddheshrainbow@gmail.com>

Rasingan commented:
@ksjc1995 
kindly go through it , will be adding html codes asap.

ksjc1995 commented on Python/Stack/implementation.py:
```diff
@@ -0,0 +1,70 @@
+        self.stack.append(element)
+
+    # we remove element from stack
+    def pop(self):
```
I think we need a check for underflow condition (i.e) when stack is empty we can't pop.

ksjc1995 replied:
`if(self.is_empty) return "Underflow";`

ksjc1995 replied:
& also i think we should return the popped element

ksjc1995 commented on Python/Stack/implementation.py:
```diff
@@ -0,0 +1,70 @@
+
+    # we don't pop the top element , just view it 
+    def top(self):
+        if self.is_empty:
```
Shouldn't this be `self.is_empty()` ? as `is_empty` is a function right?

ksjc1995 reviewed (commented):
Please add test cases as well for a variety of inputs as discussed

Rasingan committed:
-- Seperated implementation with test as discussed
-- Worked on Undeflow as asked
-- Created all the testcases using unittest python module
-- Passed three test cases checking all stack functionalities
-- Not tested is_empty , since it's already checked in pop and top operation

To run Testcases
python3 implementation_tests.py

Signed-off-by: Rasingan <siddheshrainbow@gmail.com>

Rasingan commented:
@ksjc1995 
seperated test cases as discussed.
